lib/registration.pm

```diff
@@ -255,17 +273,17 @@ sub fill_in_registration_data {
 
     # Process modules on sle 15
     if (is_sle '15+') {
-        my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
+        my $modules_needle = ("modules-preselected-" . get_required_var('SLE_PRODUCT'));
         if (check_var('BETA', '1')) {
             assert_screen('scc-beta-filter-checkbox');
             send_key('alt-i');
         }
         elsif (!check_screen($modules_needle, 0)) {
             record_info('bsc#1094457 : SLE 15 modules are still in BETA while product enter GMC phase');
-            assert_screen('scc-beta-filter-checkbox');
```

Comment:
Why you want to change this part of code? For me this assertion looks useful as if checkbox is not there we would see that with failed needle and not after when trying to fine module.
The only thing we could do is combine it with the check we do in `verify_preselected_modules`. 
Or is it the case that test fails, as needle doesn't match and checkbox is already checked?
To be honest I don't like this workaround either and with your `verify_preselected_modules` implementation we could always check it when it's there.